src/Data/Aeson/Extra/Merge.hs

```diff
@@ -51,3 +39,21 @@ mergeA :: Functor f
       => (forall a. (a -> a -> f a) -> ValueF a -> ValueF a -> f (ValueF a))
       -> Value -> Value -> f Value
 mergeA f a b = embed <$> f (mergeA f) (project a) (project b)
+
+-- | Example of using 'merge'. see <https://lodash.com/docs#merge>:
+--
+-- /Note:/ not tested against JavaScript lodash, so may disagree in the results.
+--
+-- @since 0.4.1.0
+lodashMerge :: Value -> Value -> Value
+lodashMerge = merge alg
```

Comment:
Well, the description on the lodash site contains the case on `undefined`; also JS libraries like this are normally everything but short lived (and links to their APIs even more), so I’d not introduce a dependency on that for such a simple merge algrorithm.

```haskell
-- | A simple merge, combining Values recursively and defaulting to the right hand side if it is of a different constructor.
--
-- Pseudo-code:
-- @
-- rightBiasedMerge { a: { b: 3, c: 4 } } { a: { c: 5, d: 6 } } == { a: { b: 3, c: 5, d: 6 } }
-- rightBiasedMerge [ 1 2 3 ] 4 == 4
-- rightBiasedMerge { a: 3 } [ 1 2 3 ] == [ 1 2 3 ]
rightBiasedMerge :: …
```